overlay_action.go

```diff
@@ -1,16 +1,31 @@
 package main
 
 import (
+	"fmt"
 	"path"
 )
 
 type OverlayAction struct {
-	BaseAction `yaml:",inline"`
-	Source     string
+	BaseAction  `yaml:",inline"`
+	Origin      string // origin of overlay, here the export from other action may be used
+	Source      string // external path there overlay is
+	Destination string // path inside of rootfs
 }
 
 func (overlay *OverlayAction) Run(context *DebosContext) error {
 	overlay.LogStart()
-	sourcedir := path.Join(context.recipeDir, overlay.Source)
-	return CopyTree(sourcedir, context.rootdir)
+	origin := context.recipeDir
+
+	//Trying to get a filename from exports first
+	if len(overlay.Origin) > 0 {
+		source, found := context.origins[overlay.Origin]
+		if !found {
+			return fmt.Errorf("Empty export '%s' found.", overlay.Origin)
+		}
+		origin = source
+	}
+
+	sourcedir := path.Join(origin, overlay.Source)
+
+	return CopyTree(sourcedir, path.Join(context.rootdir, overlay.Destination))
```

Comment:
todo item, ensure destination isn't causing the destination to point outside of the rootdir

Reply:
done